# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTable.cpp
# Commit: 25adf505426d
# Full Hash: 25adf505426d21f67e15d146f879b0ea53d343a7
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-03-23 04:09:13
# Regressor Bug: 1642412
# File Overlap Count: 19
# Description:
#   Bug 1642412 - wasm: Rollup of inline table.get/set for anyref. r=lth
#   
#   Rollup of the previous inline table.get/set patches.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D141024
# ==============================================================================

diff -r 43f7556db2e1 -r 25adf505426d js/src/wasm/WasmTable.cpp
--- a/js/src/wasm/WasmTable.cpp	Tue Mar 22 19:44:13 2022 +0000
+++ b/js/src/wasm/WasmTable.cpp	Tue Mar 22 19:59:05 2022 +0000
@@ -133,9 +133,9 @@
   }
 }
 
-uint8_t* Table::functionBase() const {
+uint8_t* Table::tlsElements() const {
   if (repr() == TableRepr::Ref) {
-    return nullptr;
+    return (uint8_t*)objects_.begin();
   }
   return (uint8_t*)functions_.get();
 }